manager/integration/tests/test_basic.py

```diff
@@ -729,7 +731,82 @@ def test_backup_volume_list():  # NOQA
     11. delete backup volumes(1 & 2)
     12. cleanup
     """
-    pass
+    backupstore_cleanup(client)
+
+    # create 2 volumes.
+    volume1_name, volume2_name = generate_volume_name(), generate_volume_name()
+
+    volume1 = create_and_check_volume(client, volume1_name)
+    volume2 = create_and_check_volume(client, volume2_name)
+
+    host_id = get_self_host_id()
+    volume1 = volume1.attach(hostId=host_id)
+    volume1 = common.wait_for_volume_healthy(client, volume1_name)
+    volume2 = volume2.attach(hostId=host_id)
+    volume2 = common.wait_for_volume_healthy(client, volume2_name)
+
+    bv1, backup1, snap1, _ = create_backup(client, volume1_name)
+    bv2, backup2, snap2, _ = create_backup(client, volume2_name)
+
+    def verify_no_err():
+        '''
+        request a backup list
+        verify backup list contains no error messages for volume(1,2)
+        verify backup list contains backup(1) for volume(1,2)
+        '''
+        # request a backup list
+        backup_volume_list, backup_volume_list_names = [], []
+        for i in range(100):
+            backup_volume_list = client.list_backupVolume()
+            backup_volume_list_names = [bv.name for bv in backup_volume_list]
+            if volume1_name in backup_volume_list_names \
+                    and volume2_name in backup_volume_list_names:
+                break
+            time.sleep(1)
+        assert volume1_name in backup_volume_list_names \
+               and volume2_name in backup_volume_list_names
+
+        for bv in backup_volume_list:
+            if bv.name in (volume1_name, volume2_name):
+                # verify backup list contains no error messages for volume(1,2)
+                assert len(bv['messages']) == 0
```

Comment:
Can we put this validation and the existence check in the same retry loop?

Reply:
I am not 100% understand your idea? Would you able tell me a little bit more? Which `existence check` are you referring to?

Reply:
I mean we can directly check the backups once we find the backup volume. e.g.:
```
for i in range(RETRY_COUNTS):
    verified_bv = {}
    backup_volume_list = client.list_backupVolume()
    for bv in backup_volume_list:
        if bv.name in (volume1_name, volume2_name):
            assert not bv['messages']
            backups = bv.backupList().data
            backup_names = [bv.name for bv in backups]
            if backup1.name in backup_names and backup2.name in backup_names:
                verified_bv.add(bv.name)
    if verified_bv == (volume1_name, volume2_name):
        break
    time.sleep(RETRY_INTERVAL)
assert verified_bv == set((volume1_name, volume2_name))
```
This is a template rather than the final implementation.

Reply:
Yeah! Your idea is much more elegant. Implemented! Only one difference in my implementation is that it make sure that backup1.name exists inside bv of volume1 and backup2.name exists inside bv of volume2. We do not need both backup1.name  and backup2.name to exist inside bv of a volume.